test/tests/selftest/pbs_pbstestsuite.py

```diff
@@ -0,0 +1,222 @@
+
+        # Now, change pbs.conf to turn the sched off
+        pbs_conf_val["PBS_START_SCHED"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+
+        # Start PBS again
+        self.server.pi.start()
+
+        # Verify that the scheduler didn't come up
+        self.assertFalse(self.scheduler.isUp())
+
+        # Now call revert_pbsconf() & restart PBS again
+        self.revert_pbsconf()
+
+        # Verify that the scheduler came up and start bit is 1
+        self.assertTrue(self.scheduler.isUp())
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname)
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+
+    def test_revert_pbsconf_remotemom(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf()
+        with a remote mom setup
+        """
+        remotemom = None
+        for mom in self.moms:
+            if not self.du.is_localhost(mom):
+                remotemom = mom
+                break
+        if remotemom is None:
+            self.skip_test("Test needs at least one remote Mom host,"
+                           " use -p moms=<hostname>")
+
+        pbs_conf_val = self.du.parse_pbs_config(self.server.hostname)
+        if pbs_conf_val is None or len(pbs_conf_val) < 1:
+            raise ValueError("Could not parse pbs.conf on host %s" %
+                             (server.hostname))
+
+        # Since the test requested for a mom, the server host's mom should not
+        # be ON, the rest should be set to 1
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "0")
+        self.assertEqual(pbs_conf_val["PBS_START_SERVER"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_SCHED"], "1")
+        self.assertEqual(pbs_conf_val["PBS_START_COMM"], "1")
+
+        # Check that the remote mom's pbs.conf has mom start bit on
+        pbs_conf_filepath = self.du.get_pbs_conf_file(remotemom.hostname)
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname,
+                                                pbs_conf_filepath)
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+
+        # Now set it to 0 and restart the mom
+        remotemom.pi.stop(remotemom.hostname)
+        pbs_conf_val["PBS_START_MOM"] = "0"
+        self.du.set_pbs_config(fin=pbs_conf_filepath, confs=pbs_conf_val)
+        remotemom.pi.start(remotemom.hostname)
+
+        # Confirm that the mom is down
+        self.assertFalse(remotemom.isUp())
+
+        # Now call revert_pbsconf() & restart mom again
+        self.revert_pbsconf()
+
+        # Confirm that the mom came up and start bit is 1
+        self.assertTrue(remotemom.isUp())
+        pbs_conf_val = self.du.parse_pbs_config(remotemom.hostname)
+        self.assertEqual(pbs_conf_val["PBS_START_MOM"], "1")
+
+    def test_revert_pbsconf_corelimit(self):
+        """
+        Test the functionality of PBSTestSuite.revert_pbsconf() when
+        PBS_CORE_LIMIT is reverted
```

Comment:
Maybe: 'Test the functionality of PBSTestSuite.revert_pbsconf() when PBS_CORE_LIMIT is set to a value other than the default'